ci-operator/step-registry/cucushift/pre/cucushift-pre-commands.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# prepare users
+users=""
+htpass_file=/tmp/users.htpasswd
+
+for i in $(seq 1 10);
+do
+    username="testuser-${i}"
+    password=`cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true`
+    users+="${username}:${password},"
+
+    if [ -f "${htpass_file}" ];
+    then
+        htpasswd -B -b ${htpass_file} ${username} ${password}
+    else
+        htpasswd -c -B -b ${htpass_file} ${username} ${password}
+    fi
+done
+
+# add users to cluster
+oc create secret generic htpass-secret --from-file=htpasswd=${htpass_file} -n openshift-config
+oc apply -f - <<EOF
+apiVersion: config.openshift.io/v1
+kind: OAuth
+metadata:
+  name: cluster
+spec:
+  identityProviders:
+  - name: htpassidp
+    challenge: true
+    login: true
+    mappingMethod: claim
+    type: HTPasswd
+    htpasswd:
+      fileData:
+        name: htpass-secret
+EOF
+
+# wait for oauth-openshift to rollout
+wait_auth=true
+expected_replicas=`oc get deployment oauth-openshift -n openshift-authentication -o jsonpath='{.status.replicas}'`
+while $wait_auth;
+do
+    available_replicas=`oc get deployment oauth-openshift -n openshift-authentication -o jsonpath='{.status.replicas}'`
+    if [ $expected_replicas == $available_replicas ];
+    then
+        wait_auth=false
+    else
+        sleep 3
+    fi
+done
+
+# configure cucushift runtime environment variables
+hosts=`grep server ${KUBECONFIG} | cut -d '/' -f 3 | cut -d ':' -f 1`
+ver_cli=`oc version | grep Client | cut -d ' ' -f 3`
+users=${users::-1}
+
+runtime_env=${SHARED_DIR}/runtime_env
+
+cat <<EOF >>${runtime_env}
+# cucumber setting
+export CUCUMBER_PUBLISH_QUIET=true
+
+# Export those parameters before running
+export OPENSHIFT_ENV_OCP4_HOSTS="${hosts}:lb"
+export OPENSHIFT_ENV_OCP4_ADMIN_CREDS_SPEC=${KUBECONFIG}
```

Comment:
```
     Given I switch to cluster admin pseudo user                                          # features/step_definitions/user.rb:13
      unknown credentials specification: /tmp/kubeconfig-356498767 (RuntimeError)
      /verification-tests/lib/admin_credentials.rb:96:in `get'
      /verification-tests/lib/environment.rb:113:in `admin'
      /verification-tests/lib/world.rb:124:in `admin'
      /verification-tests/features/step_definitions/user.rb:15:in `/^I switch to cluster admin pseudo user$/'
      features/routing/rate-limit.feature:8:in `I switch to cluster admin pseudo user' 
```
It should be `export OPENSHIFT_ENV_OCP4_ADMIN_CREDS_SPEC=file://${KUBECONFIG}`

Reply:
Yes, thanks, I just fixed it this morning. Then I run into

```
    Scenario: Mirroring built image doesn't degrade scheme2 ,keep consistent SHA's

Given I save a htpasswd registry auth to the :combine_dockercfg clipboard

Message:

    kubeconfig does not exists (RuntimeError)
./lib/admin_credentials.rb:77:in `get'
./lib/admin_credentials.rb:94:in `get'
```

Reply:
I think `${KUBECONFIG}` is `/tmp/kubeconfig-xxx`.
Currently using `export OPENSHIFT_ENV_OCP4_ADMIN_CREDS_SPEC=file:///${KUBECONFIG}`, so there is an extra `/` there.

Reply:
The error `kubeconfig does not exists (RuntimeError)` was found when I added `export OPENSHIFT_ENV_OCP4_ADMIN_CREDS_SPEC=file://${KUBECONFIG}`.

I updated it to `///` following the comment in `lib/admin_credentials.rb`, let's see how that works...

Reply:
Still fails, now made an update to copy ${KUBECONFIG} to `/tmp/kubeconfig` and load it from there.